### PR TITLE
Add support for reconfiguring params using GUI in ThetaStarPlanner

### PR DESCRIPTION
**Pull request by Anshu-man567**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #965 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on |  |

---

## Description of contribution in a few bullet points
- add support for reconfiguring params using GUI in ThetaStarPlanner
- removing 'w_heuristic_cost_' from Nav2 Theta Star Planner as a user configurable parameter
- updates to `nav2_theta_star_planner/README.md` to remove its reference wherever applicable
- updates to `nav2_theta_star_planner/src/theta_star_planner.cpp` to set `w_heuristic_cost` as `std::min(1.0, w_euc_cost_)`
- fixed an incorrect check of whether the start and goal pose are the same in `nav2_theta_star_planner/src/theta_star_planner.cpp`

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by SteveMacenski** [comment]
Sorry, mergify is crapping out on me right now and I'm not sure why, please ignore those comments.

File: nav2_theta_star_planner/include/nav2_theta_star_planner/theta_star_planner.hpp [hunk 1]
```diff
@@ -62,8 +64,15 @@ class ThetaStarPlanner : public nav2_core::GlobalPlanner
...
 
+  // parent node weak ptr
+  rclcpp_lifecycle::LifecycleNode::WeakPtr parent_node_;
+
   std::unique_ptr<theta_star::ThetaStar> planner_;
 
+  // Dynamic parameters handler
+  rclcpp::node_interfaces::OnSetParametersCallbackHandle::SharedPtr dyn_params_handler_;
+
+
```
Comment (SteveMacenski) [inline nav2_theta_star_planner/include/nav2_theta_star_planner/theta_star_planner.hpp:75]: Remove this excess space

File: nav2_theta_star_planner/src/theta_star_planner.cpp [hunk 2]
```diff
@@ -189,6 +194,30 @@ nav_msgs::msg::Path ThetaStarPlanner::linearInterpolation(
...
+        planner_->how_many_corners_ = parameter.as_int();
+      }
+    } else if (type == ParameterType::PARAMETER_DOUBLE) {
+      if (name == name_ + ".w_euc_cost") {
+        planner_->w_euc_cost_ = parameter.as_double();
+      } else if (name == name_ + ".w_traversal_cost") {
+        planner_->w_traversal_cost_ = parameter.as_double();
+      }
+    }
+  }
```
Comment (SteveMacenski) [inline nav2_theta_star_planner/src/theta_star_planner.cpp:216]: I think this is missing `use_final_approach_orientation`

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Just a couple small changes but LGTM

I did see that https://github.com/ros-planning/navigation2/blob/main/nav2_theta_star_planner/src/theta_star_planner.cpp#L56-L58 is here. Shouldn't that be removed as no longer a parameter?

Your description of the PR does not have some of the changes mention, did you forget to push a couple of your commits?

File: nav2_theta_star_planner/src/theta_star_planner.cpp (hunk 2 above)
Comment (Anshu-man567) [reply nav2_theta_star_planner/src/theta_star_planner.cpp:217]: fixed it

**Comment by SteveMacenski** [comment]
Your description of the PR does not have some of the changes mention, did you forget to push a couple of your commits?

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
https://github.com/ros-planning/navigation.ros.org/blob/master/migration/Galactic.rst#costmap2dros-parameters-are-now-dynamic add a section like this in the migration guide for the fact that these are now dynamic parameters

**Comment by SteveMacenski** [comment]
https://github.com/ros-planning/navigation.ros.org/blob/master/migration/Galactic.rst I just did it for you to get this moving

**Pull request merged**